pkg/task/task_manager.go

```diff
@@ -29,20 +32,55 @@ import (
 var taskManager *TaskManager
 var taskManagerOnce sync.Once
 
-func GetTaskManager() *TaskManager {
-	taskManagerOnce.Do(func() {
-		logger := log.FromContext(context.TODO())
-		taskManager = &TaskManager{
-			Logger: &logger,
+const debugTaskEnv = "DEBUG_TASK"
+
+func taskManagerInit() {
+	logger := log.FromContext(context.TODO())
+	taskManager = &TaskManager{
+		Logger: &logger,
+		tokens: make(chan struct{}, 10000),
```

Comment:
define the constant somewhere else, and it's better to be able to overwrite the value from env variable